vmware-base/src/com/cloud/hypervisor/vmware/mo/HostMO.java

```diff
@@ -1184,4 +1206,261 @@ public ManagedObjectReference waitForPortGroup(String networkName, long timeOutM
+                    // The remaining capacity of one type affects other vGPU type capacity.
+                    // Each GPU should always contain one type of vGPU VMs.
+                    if ((groupName.equals("NVIDIAGRID K1") && vgpuType.equals("grid_k100")) || (groupName.equals("NVIDIAGRID K2") && vgpuType.equals("grid_k200"))) {
+                        remainingCapacities[0] -= ((long)graphicInfo.getVm().size());
+                        remainingCapacities[1] -= 8l;
+                        remainingCapacities[2] -= 4l;
+                        remainingCapacities[3] -= 2l;
+                        remainingCapacities[4] -= 1l;
+                    } else if ((groupName.equals("NVIDIAGRID K1") && vgpuType.equals("grid_k120q")) || (groupName.equals("NVIDIAGRID K2") && vgpuType.equals("grid_k220q"))) {
+                        remainingCapacities[0] -= 8l;
+                        remainingCapacities[1] -= ((long) graphicInfo.getVm().size());
+                        remainingCapacities[2] -= 4l;
+                        remainingCapacities[3] -= 2l;
+                        remainingCapacities[4] -= 1l;
+                    } else if ((groupName.equals("NVIDIAGRID K1") && vgpuType.equals("grid_k140q")) || (groupName.equals("NVIDIAGRID K2") && vgpuType.equals("grid_k240q"))) {
+                        remainingCapacities[0] -= 8l;
+                        remainingCapacities[1] -= 8l;
+                        remainingCapacities[2] -= ((long) graphicInfo.getVm().size());
+                        remainingCapacities[3] -= 2l;
+                        remainingCapacities[4] -= 1l;
+                    } else if ((groupName.equals("NVIDIAGRID K1") && vgpuType.equals("grid_k160q")) || (groupName.equals("NVIDIAGRID K2") && vgpuType.equals("grid_k260q"))) {
+                        remainingCapacities[0] -= 8l;
+                        remainingCapacities[1] -= 8l;
+                        remainingCapacities[2] -= 4l;
+                        remainingCapacities[3] -= ((long)graphicInfo.getVm().size());
+                        remainingCapacities[4] -= 1l;
+                    } else if ((groupName.equals("NVIDIAGRID K1") && vgpuType.equals("grid_k180q")) || (groupName.equals("NVIDIAGRID K2") && vgpuType.equals("grid_k280q"))) {
+                        remainingCapacities[0] -= 8l;
+                        remainingCapacities[1] -= 8l;
+                        remainingCapacities[2] -= 4l;
+                        remainingCapacities[3] -= 2l;
+                        remainingCapacities[4] -= ((long)graphicInfo.getVm().size());
+                    }
+                }
+            }
+        }
+
+        //Pattern.matches("grid_k100|grid_k1[2468]0[q]", gpuType)
+        for (final String gpuType : sharedPassthruGpuTypes) {
+            if ((groupName.equals("NVIDIAGRID K1") && gpuType.equals("grid_k100")) || (groupName.equals("NVIDIAGRID K2") && gpuType.equals("grid_k200"))) {
```

Comment:
@nitin-maharana it's probably worth exploring refactoring each supported gfx card type as it's own class wrapping the behaviour of what gpuCapacity to put?